### DISCUSSION TITLE
How to get a usable blob from PIXI.Extract?

### DISCUSSION THREAD

**Discussion by manvankaliber** [pixijs/pixijs#7955] [Q&A]
Hi everyone,

I want to create a new blob url from my canvas using PIXI.Extract so I can `URL.createObjectURL(blob)` it for my `<img>` as its source. Is this possible with PIXI.Extract? 

In the docs I only spot the base64 function and pixels function. I'm not sure if I can use the raw pixels array from the pixels function to create a blob. Anyone have any experience with this?

I basically want the behavior from `canvas.toBlob()` when using the WebGL renderer..

**Comment by jmlee2k** [comment]
Try something like this:

```javascript
render.plugins.extract.canvas(container).toBlob(b=>{
    //do something with b here
});
```

where `container` is the stage (or any other container) you want to extract from.

**Comment by manvankaliber** [comment]
Doesn't this just run the toBlob function tot he original canvas? That function isn't available for webgl contexts..

**Reply by jmlee2k** [reply]
I'm not sure of what it does in the background, but it works in our code base (we use it to capture textures for debugging).

**Comment by bigtimebuddy** [comment]
If you need the Blob, you could try to use `fetch`:

```js
const dataUri = renderer.plugins.extract.base64(target);
const response = await fetch(dataUri);
const blob = await response.blob();
```

Alternatively, if you just need the image, why not:

```js
const img = renderer.plugins.extract.image(target);
```

Example: https://jsfiddle.net/bigtimebuddy/3gk0a7v5/
